### PR TITLE
Fix A2D3 not rendering on Firefox

### PR DESCRIPTION
**Pull request by jfarseneau**
All A2D3 rendered charts were not showing up in the latest version of Firefox. This restores A2D3 charts for Firefox by using the parent's width and adding in any padding if necessary for width and height.

This, slightly out of the intended scope but necessary to make Circle happy, also updates outdated karma and webpack related packages and changes testing to use *.spec.ts files in the src/angularD3 directory.

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // @AngularClass
 
-module.exports = function(config) {
+module.exports = function (config) {
```
Comment (chrisnicola) [inline karma.conf.js:3]: Why the space here?

**Review by chrisnicola — APPROVED** [review]
Looks good.

File: karma.conf.js (hunk 1 above)
Comment (jfarseneau) [reply karma.conf.js:3]: Likely strange behaviour from my auto-document formatter. I will go strip it out and up the a2d3 version.

Comment (jfarseneau) [reply karma.conf.js:3]: Nevermind, noticed the a2d3 version is up. Thanks for reviewing and deploying!

**Pull request merged**